### ISSUE TITLE
Verify compliance

### ISSUE THREAD

**Issue by robgpeak** [robgpeak/CL-WebExtension#18]
@AddisonFreeman Please have a read of https://pubhelp.rakutenmarketing.com/hc/en-us/articles/200965808-Downloadable-Software-Applications-DSAs-An-Overview and make sure we are in compliance with their requirements for the extension.

**Comment by AddisonFreeman** [comment]
https://pubhelp.rakutenmarketing.com/hc/en-us/article_attachments/202932599/downloadable_service_agreement_2-2-15-v2PDF.pdf

I'm looking at the Development & Functional Requirements section. 

- DSA’s should be constructed with an opt-in/opt-out option for advertisers who are
affiliated with the publisher but do not want to be a part of the DSA.
- DSA’s should be constructed with an opt-in/opt-out option for consumers.
- The DSA must not force clicks or “cookie stuff”. The DSA must not insert a cookie onto
the user’s computer without the user knowingly taking an action that results in the
cookie being placed.
*** I just pushed (1.3.27) a change where a timestamp cookie is only saved after the user clicks the activate button, before it was done automatically. The extension also saves a user's email address, points, and the primary and accent colors of the subdomain in the extension storage area. Don't know if that applies here, it's technically not a cookie ***

There's more points in the pdf I just wanted your thoughts on the opt-in/out consent and cookies.

Other than the above, they need to do compliance testing on the extension, I'm sure you were aware, I didn't see anything on how long that would take.

**Comment by AddisonFreeman** [comment]
There's also this under the 'Guidelines and Functional Requirements For End User Notification' section:

- Notifications which prompt the user to take action should be prompted only once per session. 
*** Ebates and Southwest's RR extension doesn't do this, their notification (and ours) appears until you click 'activate later' or x out of it. ***

**Comment by robgpeak** [comment]
Just get it as close to compliance as you can, and we will submit it and
see what they say.

Thanks,

Rob

**Sent from my phone

**Comment by robgpeak** [comment]
@AddisonFreeman We are going to need to build in a feature/button that allows people to "opt-out".  If you look to see how some of the other extensions do it, you can emulate that.  This is point number 10 in their list of what we need to submit to them for approval.  It is on the page of the first link in this thread.

**Comment by cgladue** [comment]
Why wouldn’t the “opt out” mechanism be just disabling or uninstalling the extension?

**Comment by robgpeak** [comment]
LOL  Chad, you're writing this like you've never dealt with LinkShare/Rakuten.  Some things they do don't make sense, but they have their reasons...

**Comment by cgladue** [comment]
You mean since they own eBay eats they’re doing it to stay for competition?

**Comment by robgpeak** [comment]
seemingly:)

**Comment by robgpeak** [comment]
@AddisonFreeman  I just got off the phone with Rakuten (Link Share) support as they we trying to log into the browser extension, and she could not get it.  She went as far as setting un an account on shop.complinks.co, and I can actually see she successfully created an account, but she said when she clicks the setting button, the button turns black but it does not take her to the screen to enter her email address.

Also, she asked if it was possible to give them a link to the developer mode of the extension on the chrome webstore, so we don;t have to send several zip files to her and her team?

For a bit more context, I am going to send you both the actual email I sent to her with the instructions on how to test the DSA including the screen shots...

![capture](https://user-images.githubusercontent.com/28393235/39490485-078dce7c-4d57-11e8-829f-22a5fabba8a8.PNG)

**Comment by robgpeak** [comment]
Actually Addison, I just realized I do not have your email address...

**Comment by AddisonFreeman** [comment]
afree1423@gmail.com

**Comment by AddisonFreeman** [comment]
That is very odd, I've never seen that happen with the settings button.

**Comment by robgpeak** [comment]
I'm not sure what to tell her?

**Comment by AddisonFreeman** [comment]
You can give her my number or skype (210)382-5214. She may have an old
version of chrome?

**Comment by robgpeak** [comment]
Addison,

I just got this email back from Regina at Rakuten...

Hi Rob,

Based on my initial testing, I’m seeing  another issue.

Your software is not standing down on another Publishers traffic when I  visited Walmart through a voucher site: Slickdeals.  See attached screenshot.  We do not allow it, so please fix this issue.  I have attached our Software agreement. (below)  Page 3 outlines RAN parameters in URL that will help you detect and stand down for traffic generated from another publishers.  Page 4 outlines the functional requirements of publishers’ software.

 Here are the list of parameters.  

Linksynergy and click.linksynergy (11/9/2010)
afsrc=1 (11/9/2010)
affiliate.rakuten.com (3/15/2013)

Please go ahead and fix the above issue. Once you are done, please let me know and I can resume testing. 

Sincerely,

Regina Cotto
MANAGER, NETWORK QUALITY

---

So if traffic has been directed to an advertiser's site from someone else's link, we must "stand down" and NOT show our popup.
![capture 1](https://user-images.githubusercontent.com/28393235/39546769-881ee18a-4e23-11e8-8056-95165004b531.PNG)
[downloadable_service_agreement_2-2-15-v2PDF (1).pdf](https://github.com/robgpeak/CL-WebExtension/files/1968733/downloadable_service_agreement_2-2-15-v2PDF.1.pdf)

**Comment by cgladue** [comment]
i wonder if eBates DSA will standdown when we direct traffic to walmart ?!



Chad Gladue
*DevNull Technology, LLC*

**Comment by AddisonFreeman** [comment]
I understand now what url params to read to not show the notification for
(on traffic from other sites and a complinks subdomain) I'll have a patch
tomorrow.

**Comment by robgpeak** [comment]
Actually I think it does.

Thanks,

Rob

**Sent from my phone

**Comment by AddisonFreeman** [comment]
@cgladue @robgpeak I have this working now on 1.4.3 (just pushed). I'm checking for that parameter on any current url or link clicked on and disabling the notification on any pages redirected to in the following 5 seconds. (Most pages keep those urls so the delay isn't always necessary but some retailers do a redirect after the initial redirect.

**Comment by AddisonFreeman** [comment]
![ebatesstanddowndemo](https://user-images.githubusercontent.com/5950267/39598544-7f0d8478-4ede-11e8-8e1a-6c0341921946.gif)

You may need to update your chrome? or double check that it loaded properly, it's working on my computer.

**Comment by robgpeak** [comment]
Ok.

**Comment by AddisonFreeman** [comment]
I pushed a minor update, shouldn't behave any differently, but maybe there was a syntax error. check for 1.4.4 now.

**Comment by robgpeak** [comment]
I just cleared my cache uninstall the old extension installed the new one 1.4.4 and now our notification is not popping up on any site that I go to...  Nevermind standing down, it's not standing up in the first place.  https://screencast.com/t/hLhIAjdQF

**Comment by AddisonFreeman** [comment]
Okay, try 1.4.5 I just pushed. If that doesn't work for you let's do a screen share.

**Comment by AddisonFreeman** [comment]
<img width="590" alt="screen shot 2018-05-03 at 2 49 37 pm" src="https://user-images.githubusercontent.com/5950267/39599358-3da6d040-4ee1-11e8-91c9-c2d73c8b7e26.png">
also make sure you're up to date

**Comment by robgpeak** [comment]
Addison, it still does not work, and it MUST work in older browsers versions.

**Comment by AddisonFreeman** [comment]
I've been testing in my virtual machine. I'll verify that my changes work
on versions 63 - 66 in that environment before I give you a version to send
back to Rakuten. I'll also check in with ideas for a catch-all stand down
solution.

**Comment by robgpeak** [comment]
@cgladue I wanted to bring you back in the loop on this issue of "Standing Down" to all other affiliate traffic.  We need to get this resolved ASAP and we need your gray matter on the case to figure out a solution.

**Comment by AddisonFreeman** [comment]
@cgladue Right now the extension will stand down to url params: afsrc === "1", linksynergy, ebtoken, wmlspartner, and affiliate.rakuten.com.

**Comment by robgpeak** [comment]
I am not in front of my computer at the moment, but I need you to test the shit out of it to make triple sure it stands down as it should in all cases before I re-submit it to Rakuten

**Comment by AddisonFreeman** [comment]
Sent you a video on Skype, Rob.
